src/eth_plugin_interface.h

```diff
@@ -55,167 +54,405 @@ typedef enum eth_plugin_result_e {
 
-    uint8_t *tokenLookup1;  // set by the plugin if a token should be looked up
-    uint8_t *tokenLookup2;
+    // RW INPUT. Contains the semi-persistent RAM space that can be used by the
+    // plugin in each handle call.
+    uint8_t *pluginContext;
 
-    const uint8_t *amount;   // set an uint256 pointer if uiType is UI_AMOUNT_ADDRESS
-    const uint8_t *address;  // set to the destination address if uiType is UI_AMOUNT_ADDRESS. Set
-                             // to the user's address if uiType is UI_TYPE_GENERIC
+    // OUTPUT. The plugin can set this value to a 20 bytes array in
+    // pluginContext containing the address of an ERC20 token. In this case
+    // Ethereum will call the plugin with handle_provide_token() with the
+    // requested ERC20 token information. The Ethereum application must be made
+    // aware of ERC20 token information first using 'PROVIDE ERC 20 TOKEN
+    // INFORMATION' APDU. Leave the value at NULL if not needed
+    uint8_t *tokenLookup1;
+    // OUTPUT. Same as tokenLookup1.
+    uint8_t *tokenLookup2;
 
+    // OUTPUT. The plugin needs to set this pointer to a 256 bits number in
+    // pluginContext to display as the amount in UI_AMOUNT_ADDRESS case. IGNORED
+    // if uiType is UI_AMOUNT_ADDRESS.
+    const uint8_t *amount;
+    // OUTPUT. The plugin needs to set this pointer to a 20 bytes address in
+    // pluginContext to display as the address in UI_AMOUNT_ADDRESS case.
+    // IGNORED if uiType is UI_AMOUNT_ADDRESS.
+    const uint8_t *address;
+
+    // OUTPUT. The plugin needs to set this value to either
+    // ETH_UI_TYPE_AMOUNT_ADDRESS for an amount/address UI or
+    // ETH_UI_TYPE_GENERIC for a generic UI.
     eth_ui_type_t uiType;
-    uint8_t numScreens;  // ignored if uiType is UI_AMOUNT_ADDRESS
+
+    // OUTPUT. The plugin needs to set this value to the number of screens
+    // needed to display the smart contract in ETH_UI_TYPE_GENERIC case. IGNORED
+    // if uiType is UI_AMOUNT_ADDRESS.
+    uint8_t numScreens;
+
+    // OUTPUT. Used by the plugin to inform the Ethereum application of the
+    // result of this handle The following return codes are expected, any other
+    // will abort the signing process:
+    // - ETH_PLUGIN_RESULT_OK
+    // - ETH_PLUGIN_RESULT_FALLBACK : if the signing logic should fallback to
+    // the generic one
     eth_plugin_result_t result;
 
 } ethPluginFinalize_t;
-// void handle_finalize(ethPluginFinalize_t *parameters);
+// --8<-- [end:handle_finalize_parameters]
+
+/*
+ --8<-- [start:handle_provide_token_explanation]
+The Ethereum application will call this handle when the plugin has requested
+ERC20 token information in `handle_finalize`
+
+The plugin needs to perform the following actions:
+
+- Check that provided token matches the one requested.
+- Increase the number of screens needed for the display if needed.
+- Report an error to the ethereum application if the display must be aborted.
 
+!!! warning
```

Comment:
The formating seems wrongly rendered.... Only the 1st line is in the box

Reply:
Fixed in this version